### PR TITLE
Add Rt and Gt reading functions to FREDJob

### PR DESCRIPTION
**Pull request by lanecodes**
Add `get_job_gt_table` and `get_job_rt_table` methods to `FREDJob`. These methods support reading Rt and Gt variables that are automatically generated for us by FRED for transmissible conditions.

To support these changes we also refactor `FREDJob` to extract common file reading functionality. Adds private methods `_read_plot_data_file`, `_convert_wide_plot_table_to_long`, and `_validate_table_format` to `FREDJob` which encapsulate the logic for reading and processing data from the `JOB/<n>/OUT/PLOT/<interval>` directory. These are now called by `get_job_variable_table` and `get_job_state_table`.

### CODE REVIEW

**Review by edzwick — APPROVED** [review]

File: epxresults/job.py [hunk 1]
```diff
@@ -273,54 +278,45 @@ def get_job_state_table(
...
-        elif format == 'wide':
-            # do nothing
-            pass
-        else:
-            msg = (f"Requested DataFrame format, {format}, not recognized.")
-            raise ValueError(msg)
+        df = self._read_plot_data_file(path_to_file, interval)
+        self._validate_table_format(format)
+        if format == "long":
+            df = self._convert_wide_plot_table_to_long(df, count_type)
```
Comment (duncandc) [inline epxresults/job.py:310]: This is tidier.  Good change.

**Review by clairemackay — APPROVED** [review]

File: epxresults/job.py [hunk 2]
```diff
@@ -273,54 +278,45 @@ def get_job_state_table(
...
 
     def get_job_variable_table(
-            self,
-            variable: str,
-            interval: str = 'daily',
-            format: str = 'long'
-            ) -> pd.DataFrame:
+        self,
+        variable: str,
+        interval: OutputInterval = "daily",
```
Comment (duncandc) [inline epxresults/job.py:317]: Why are double quotes needed/suggested here?

File: epxresults/job.py [hunk 3]
```diff
@@ -364,36 +360,179 @@ def get_job_variable_table(
...
+        See `FRED documentation`_ for further details of the definition of this
+        quantity.
+
+        .. _FRED documentation: https://docs.epistemix.com/projects/fred-sims/en/latest/sim-guide/chapter2.html#reproductive-rate-rt
+
+        Examples
+        --------
+        >>> job = FREDJob(job_key='simpleflu')
+        >>> df = job.get_job_rt_table('INF').head()
+        >>> df  # doctest: +SKIP
```
Comment (duncandc) [inline epxresults/job.py:397]: I assume `# doctest: +SKIP` makes pytest skip this line?  If so, this is excellent!

**Review by duncandc — APPROVED** [review]
The addition of Rt and Gt table retrieval is a needed improvement. In addition, this PR makes some nice improvements in the code organization and utilities associated with the job class.

**Comment by duncandc** [comment]
It seems odd that the job class can now access this data, but the run class cannot.  I know we have discussed and fretted about duplicating functionality between the `FREDJob` and `FREDRun` classes before. But I wanted to make a note here of this divergence.

File: epxresults/job.py (hunk 2 above)
Comment (lanecodes) [reply epxresults/job.py:317]: They're not strictly necessary, but 
1. Double quotes are used in other parts of this function definition, and consistency is good
2. I am in the habit of running [black](https://github.com/psf/black) over changes I make to ensure removal of unnecessary whitespace etc, and black uses double quotes.

File: epxresults/job.py (hunk 3 above)
Comment (lanecodes) [reply epxresults/job.py:397]: Yes, that's it. The issue here is that I can't guarantee identical numerical outputs when someone else runs the scripts to generate the test data. This lets us show the structure of the generated table in the docstring without needing to worry about making such assurances.

**Comment by lanecodes** [comment]
@duncandc yes, I see your point but think it's going to slow down development if we need to effectively solve the same problem twice for each piece of new functionality.

**Pull request merged**